src/backend/distributed/utils/background_jobs.c

```diff
@@ -758,15 +811,25 @@ TaskEnded(TaskExecutionContext *taskExecutionContext)
 void
 CitusBackgroundTaskQueueMonitorMain(Datum arg)
 {
+	/* handle SIGTERM to properly kill active task executors */
+	pqsignal(SIGTERM, QueueMonitorSigTermHandler);
+
+	/* ignore SIGINT signal */
+	pqsignal(SIGINT, SIG_IGN);
```

Comment:
Why is this necessary? What happens otherwise? This is not done for the maintenance daemon. Should we do it there too?

Reply:
SIGINT is sent by pg_backend_cancel. But, we donot want to cancel queue monitor upon recieving the signal.
Default handler for bgworker which runs a backend is like below:

```
if (worker->bgw_flags & BGWORKER_BACKEND_DATABASE_CONNECTION)
{
       /*
	* SIGINT is used to signal canceling the current action
	*/
       pqsignal(SIGINT, StatementCancelHandler);
```